Kitty_Paw_Keypad/code.py

```diff
@@ -52,89 +57,62 @@
 #  display setup
 display = ST7789(display_bus, width=240, height=240, rowstart=80)
 
-# CircuitPython 6 & 7 compatible
-#  bitmap setup
-bitmap = displayio.OnDiskBitmap(open("/parrot-240-sheet.bmp", "rb"))
-
-# Create a TileGrid to hold the bitmap
-parrot0_grid = displayio.TileGrid(bitmap, pixel_shader=getattr(bitmap, 'pixel_shader', displayio.ColorConverter()),
-                                 tile_height=240, tile_width=240)
-
-# # CircuitPython 7+ compatible
-# bitmap = displayio.OnDiskBitmap("/parrot-240-sheet.bmp")
+bitmap, palette = adafruit_imageload.load("/partyParrotsSmol.bmp",
+                                          bitmap=displayio.Bitmap,
+                                          palette=displayio.Palette)
 
 # Create a TileGrid to hold the bitmap
-# parrot0_grid = displayio.TileGrid(bitmap, pixel_shader=bitmap.pixel_shader,
-#                                tile_height=240, tile_width=240)
+parrot0_grid = displayio.TileGrid(bitmap, pixel_shader=palette,
+                                 tile_height=32, tile_width=32)
 
 # Create a Group to hold the TileGrid
-group = displayio.Group()
+group = displayio.Group(scale=4, x = 64, y = 32)
 
 # Add the TileGrid to the Group
 group.append(parrot0_grid)
 
 # Add the Group to the Display
 display.show(group)
 
-#  digital pins for the buttons
-key_pins = [board.A0, board.A1, board.A2, board.A3]
-
-#  array for buttons
-keys = []
+#  setup button pins
+key_pins = (
+    board.A0,
+    board.A1,
+    board.A2,
+    board.A3,
+)
 
-#  setup buttons as inputs
-for key in key_pins:
-    key_pin = digitalio.DigitalInOut(key)
-    key_pin.direction = digitalio.Direction.INPUT
-    key_pin.pull = digitalio.Pull.UP
-    keys.append(key_pin)
+#  create keypad
+keys = keypad.Keys(key_pins, value_when_pressed=False, pull=True)
 
 p = 0 #  variable for tilegrid index
-a = 0 #  variable for tile position
-
-#  states for buttons
-key0_pressed = False
-key1_pressed = False
-key2_pressed = False
-key3_pressed = False
-
-#  array for button states
-key_states = [key0_pressed, key1_pressed, key2_pressed, key3_pressed]
 
-last_p = p
 while True:
-    #  default tile grid position
-    if last_p != p:
-        parrot0_grid[a] = p
-        last_p = p
-
-    #  iterate through 4 buttons
-    for i in range(4):
-        inputs = keys[i]
-        #  if button is pressed...
-        if not inputs.value and key_states[i] is False:
-            #  tile grid advances by 1 frame
+    #  update parrot sprite
+    parrot0_grid[0] = p
```

Comment:
This is changing the grid every time around the loop. It might be a no-op, but maybe moving it inside the if and adjusting the wrapping increment location as well, as below, would be showing good practice.
```suggestion
```